pkg/build/builder/docker.go

```diff
@@ -318,6 +319,13 @@ func (d *DockerBuilder) dockerBuild(dir string, tag string, secrets []api.Secret
 		if d.build.Spec.Strategy.DockerStrategy.DockerfilePath != "" {
 			dockerfilePath = d.build.Spec.Strategy.DockerStrategy.DockerfilePath
 		}
+		// TODO: Check for non-existent build args here, if necessary
```

Comment:
Currently, if you pass a build arg to Docker which doesn't get used, the build fails. This is only reported in the Build logs, which may be a bit harder to debug. We could, alternatively, check for build args when we parse the Dockerfile, but this may be more complicated than we want.